---
title: “LDAP身份验证失败并出现超时错误 | AEM Oak”
description: 描述
solution: Experience Manager
product: Experience Manager
applies-to: "Experience Manager"
keywords: “KCS、ldap、aem、超时、登录、身份验证”
resolution: Resolution
internal-notes: null
bug: false
article-created-by: Amol Mahajan
article-created-date: "7/3/2023 2:09:40 PM"
article-published-by: Amol Mahajan
article-published-date: "7/3/2023 2:12:29 PM"
version-number: 5
article-number: KA-17498
dynamics-url: "https://adobe-ent.crm.dynamics.com/main.aspx?forceUCI=1&pagetype=entityrecord&etn=knowledgearticle&id=fe7a2b3e-ab19-ee11-8f6e-6045bd006295"
source-git-commit: 8578ebc08751e921d354a74b5c0e9e1a4dbbe7cb
workflow-type: tm+mt
source-wordcount: '440'
ht-degree: 2%

---

# LDAP身份验证失败并出现超时错误 |AEM Oak

## 描述 {#description}


### <b>环境</b>

Adobe Experience Manager



### <b>问题/症状</b>

配置后 [通过AEM进行LDAP身份验证](https://experienceleague.adobe.com/docs/experience-manager-65/administering/security/ldap-config.html?lang=en)，无法允许LDAP用户登录。 将显示以下日志消息：


```
Caused by: org.apache.directory.api.ldap.model.exception.LdapException: TimeOut occurred

at org.apache.directory.ldap.client.api.LdapNetworkConnection.writeRequest(LdapNetworkConnection.java:4106)

at org.apache.directory.ldap.client.api.LdapNetworkConnection.bindAsync(LdapNetworkConnection.java:1290)

at org.apache.directory.ldap.client.api.LdapNetworkConnection.bind(LdapNetworkConnection.java:1188)

at org.apache.directory.ldap.client.api.AbstractLdapConnection.bind(AbstractLdapConnection.java:127)

at org.apache.directory.ldap.client.api.AbstractLdapConnection.bind(AbstractLdapConnection.java:112)

at org.apache.directory.ldap.client.api.DefaultLdapConnectionFactory.bindConnection(DefaultLdapConnectionFactory.java:64)

at org.apache.directory.ldap.client.api.DefaultLdapConnectionFactory.newLdapConnection(DefaultLdapConnectionFactory.java:107)

at org.apache.directory.ldap.client.api.ValidatingPoolableLdapConnectionFactory.makeObject(ValidatingPoolableLdapConnectionFactory.java:133)

at org.apache.directory.ldap.client.api.ValidatingPoolableLdapConnectionFactory.makeObject(ValidatingPoolableLdapConnectionFactory.java:59)

at org.apache.commons.pool.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:1188)

at org.apache.directory.ldap.client.api.LdapConnectionPool.getConnection(LdapConnectionPool.java:123)

at org.apache.jackrabbit.oak.security.authentication.ldap.impl.LdapIdentityProvider.connect(LdapIdentityProvider.java:771)

... 57 common frames omitted
```



## 解决方法 {#resolution}


要解决此问题，您需要调试从AEM到LDAP服务器的连接。 以下是一些要检查的内容：

1. 检查LDAP服务器是否可通过AEM服务器以外的其他计算机访问，例如 [JXplorer](https://jxplorer.org/). 如果无法访问该服务器，则可能是该服务器处于关闭状态，或者可能存在网络或防火墙问题。 请联系您的网络运营团队和管理您的LDAP服务器的团队进行调查。
2. 如果LDAP服务器可从其他计算机访问，则从AEM服务器操作系统进行测试。 在AEM服务器操作系统上安装LDAP客户端，并尝试从此处访问LDAP服务器。 在Linux上，您可以使用 [ldapsearch](https://access.redhat.com/documentation/en-us/red_hat_directory_server/11/html/administration_guide/examples-of-common-ldapsearches) 命令。 在Windows上，使用 [JXplorer](https://jxplorer.org/).
3. 如果服务器可以访问LDAP服务器，但是基于AEM LDAP的登录失败，那么我们需要检查 <b>LDAP身份提供程序</b> 配置。 登录到 <b>OSGi Web控制台</b> (https://*aem-host：port*/system/console/configMgr)并搜索 <b>Apache Jackrabbit Oak LDAP身份提供程序</b>. 您可以尝试一些可能解决此问题的方法：

   - 微调 *用户基础DN*， *用户额外筛选器*， *组基本DN*、和 *对额外过滤器进行分组* 使搜索过滤器只将相关用户和组返回到AEM。
   - 确保 *绑定DN* 和 *绑定密码* 是正确的。
   - 取消选中 *验证时查找管理池* 和 *验证时查找用户池。*
   - 增加 *搜索超时。*

   <b>LDAP身份提供程序配置的屏幕快照：</b>


   ![rtaimage_3_](https://helpx.adobe.com/content/dam/help/en/experience-manager/kb/LDAP-error/jcr%3acontent/main-pars/image/rtaimage_3_.png "rtaimage_3_")
4. 对于大多数企业客户来说， LDAP通常是负载平衡的。 如果位于LDAP服务器前面的负载平衡器由于某种原因将您的AEM服务器IP列入黑名单，则您也可能会遇到此问题。 如果出现此问题，请联系LDAP团队以解决此问题。 作为快速测试，您可能希望直接绕过LDAP负载平衡器来点击LDAP服务器IP，以查看AEM中的LDAP身份验证是否成功。


### <b>原因</b>

超时错误通常表示AEM由于防火墙、网络问题而无法访问LDAP服务器，或者该服务器已关闭或无响应。
